py2vega fails on Python 3.9

We're seeing issues on test suites in the `ipydatagrid` repo when using Python 3.9. 

The error seems to originate in `py2vega` : "Unsupported Constant node, note that only a subset of Python is supported". 

Trace available below:

https://github.com/bloomberg/ipydatagrid/pull/191/checks?check_run_id=1991433011

@martinRenou perhaps you could point me in the right direction? Thanks :)

Seems to be happening using the py2vega tests in this repo:


```bash
test/test_py2vega.py ...........F......................                                                          [100%]

======================================================= FAILURES =======================================================
____________________________________________________ test_subscript ____________________________________________________

    def test_subscript():
        code = 'value[0]'
>       assert py2vega(code, whitelist) == 'value[0]'

test/test_py2vega.py:137:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
py2vega/main.py:340: in py2vega
    return VegaExpressionVisitor(whitelist).visit(parsed.body)
../../miniconda3/envs/py2vega/lib/python3.9/ast.py:407: in visit
    return visitor(node)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <py2vega.main.VegaExpressionVisitor object at 0x7f0c22048880>, node = <ast.Subscript object at 0x7f0c22048a30>

    def visit_Subscript(self, node):
        """Turn a Python Subscript node into a Vega-expression."""
        value = self.visit(node.value)

        if isinstance(node.slice, ast.Index):
            return '{value}[{index}]'.format(
                value=value,
                index=self.visit(node.slice.value)
            )

        if isinstance(node.slice, ast.Slice):
            if node.slice.step is not None:
                raise Py2VegaSyntaxError('Unsupported step for {} node'.format(node.slice.__class__.__name__))

            args = [value, '0' if node.slice.lower is None else self.visit(node.slice.lower)]
            if node.slice.upper is not None:
                args += self.visit(node.slice.upper)

            return 'slice({args})'.format(args=', '.join(args))

>       raise Py2VegaSyntaxError('Unsupported {} node'.format(node.slice.__class__.__name__))
E         File "<string>", line None
E       py2vega.main.Py2VegaSyntaxError: Unsupported Constant node, note that only a subset of Python is supported

py2vega/main.py:323: Py2VegaSyntaxError
=================================================== warnings summary ===================================================
test/test_py2vega.py::test_nameconstant
test/test_py2vega.py::test_nameconstant
test/test_py2vega.py::test_nameconstant
test/test_py2vega.py::test_tuple
test/test_py2vega.py::test_tuple
test/test_py2vega.py::test_list
  /home/idafna/miniconda3/envs/py2vega/lib/python3.9/ast.py:407: DeprecationWarning: visit_NameConstant is deprecated; add visit_Constant
    return visitor(node)

test/test_py2vega.py: 38 warnings
  /home/idafna/miniconda3/envs/py2vega/lib/python3.9/ast.py:407: DeprecationWarning: visit_Num is deprecated; add visit_Constant
    return visitor(node)

test/test_py2vega.py: 32 warnings
  /home/idafna/miniconda3/envs/py2vega/lib/python3.9/ast.py:407: DeprecationWarning: visit_Str is deprecated; add visit_Constant
    return visitor(node)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=============================================== short test summary info ================================================
FAILED test/test_py2vega.py::test_subscript -   File "<string>", line None
====================================== 1 failed, 33 passed, 76 warnings in 0.18s =======================================
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -20,7 +20,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        python-version: [3.7, 3.8]
+        python-version: [3.6, 3.7, 3.8, 3.9]
 
     steps:
     - name: Checkout

--- a/py2vega/main.py
+++ b/py2vega/main.py
@@ -1,7 +1,6 @@
 """Python to VegaExpression transpiler."""
 
 import ast
-import sys
 
 import inspect
 import types
@@ -264,14 +263,6 @@ class VegaExpressionVisitor(ast.NodeVisitor):
 
     def visit_Name(self, node):
         """Turn a Python name expression into a Vega-expression."""
-        if sys.version_info[0] == 2:
-            if node.id == 'False':
-                return 'false'
-            if node.id == 'True':
-                return 'true'
-            if node.id == 'None':
-                return 'null'
-
         # If it's in the scope, return it's evaluated expression
         if node.id in self.scope:
             return self.scope[node.id]
@@ -304,12 +295,6 @@ class VegaExpressionVisitor(ast.NodeVisitor):
         """Turn a Python Subscript node into a Vega-expression."""
         value = self.visit(node.value)
 
-        if isinstance(node.slice, ast.Index):
-            return '{value}[{index}]'.format(
-                value=value,
-                index=self.visit(node.slice.value)
-            )
-
         if isinstance(node.slice, ast.Slice):
             if node.slice.step is not None:
                 raise Py2VegaSyntaxError('Unsupported step for {} node'.format(node.slice.__class__.__name__))
@@ -320,7 +305,14 @@ class VegaExpressionVisitor(ast.NodeVisitor):
 
             return 'slice({args})'.format(args=', '.join(args))
 
-        raise Py2VegaSyntaxError('Unsupported {} node'.format(node.slice.__class__.__name__))
+        return '{value}[{index}]'.format(
+            value=value,
+            index=self.visit(node.slice)
+        )
+
+    def visit_Index(self, node):
+        """Turn a Python subscript index into a Vega-expression."""
+        return self.visit(node.value)
 
     def visit_Attribute(self, node):
         """Turn a Python attribute expression into a Vega-expression."""

--- a/setup.py
+++ b/setup.py
@@ -14,7 +14,7 @@ setup(
     license='BSD 3-Clause',
     keywords='python vega vega-expression',
     packages=find_packages(exclude=['test']),
-    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*',
+    python_requires='>=3.6',
     install_requires=[],
     extras_require={
         'testing': ['pytest', 'flake8'],
@@ -23,13 +23,10 @@ setup(
     classifiers=[
         'Intended Audience :: Developers',
         'Operating System :: OS Independent',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.4',
-        'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: 3.9',
         'Topic :: Software Development :: Libraries :: Python Modules',
     ],
 )

